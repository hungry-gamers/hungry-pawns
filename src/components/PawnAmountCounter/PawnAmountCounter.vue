<script setup lang="ts">
import type { PawnSize } from '@/stores/game/types.ts'
import { computed } from 'vue'

const props = defineProps<{
  size: PawnSize
  amount: number
  isIncreaseDisabled: boolean
}>()

const emit = defineEmits(['on-amount-changed'])

const isDecreaseDisabled = computed(() => props.amount === 0)

const decrease = () => {
  emit('on-amount-changed', { amount: props.amount - 1, size: props.size })
}

const increase = () => {
  emit('on-amount-changed', { amount: props.amount + 1, size: props.size })
}
</script>

<template>
  <div>
    <button @click="decrease" :disabled="isDecreaseDisabled">-</button>
    <span>{{ amount }}</span>
    <button @click="increase" :disabled="isIncreaseDisabled">+</button>
  </div>
</template>

<style scoped></style>
