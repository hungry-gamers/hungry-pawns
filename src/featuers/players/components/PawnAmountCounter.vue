<script setup lang="ts">
import type { PawnSize } from '@/featuers/game/types.ts'
import { computed } from 'vue'

const props = defineProps<{
  size: PawnSize
  amount: number
  isIncreaseDisabled: boolean
}>()

const emit = defineEmits(['on-amount-changed'])

const isDecreaseDisabled = computed(() => props.amount === 0)

const decrease = () => {
  emit('on-amount-changed', { amount: props.amount - 1, size: props.size })
}

const increase = () => {
  emit('on-amount-changed', { amount: props.amount + 1, size: props.size })
}
</script>

<template>
  <div>
    <span>{{ size }}</span>
    <button @click="decrease" :disabled="isDecreaseDisabled" data-test-id="decrease-button">
      -
    </button>
    <span>{{ amount }}</span>
    <button @click="increase" :disabled="isIncreaseDisabled" data-test-id="increase-button">
      +
    </button>
  </div>
</template>

<style scoped></style>
