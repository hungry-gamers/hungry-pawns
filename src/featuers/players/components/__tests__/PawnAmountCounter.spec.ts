import { describe, it, expect, beforeEach, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import PawnAmountCounter from '../PawnAmountCounter.vue'

describe('PawnAmountCounter', () => {
  const wrapper = mount(PawnAmountCounter, {
    props: {
      size: 'small',
      amount: 2,
      isIncreaseDisabled: false,
    },
  })

  beforeEach(() => vi.clearAllMocks())

  it('should render properly', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should emit decrease event', () => {
    const button = wrapper.find('[data-test-id="decrease-button"]')
    button.trigger('click')
    expect(wrapper.emitted()['on-amount-changed']).toContainEqual([
      {
        amount: 1,
        size: 'small',
      },
    ])
  })

  it('should emit increase event', () => {
    const button = wrapper.find('[data-test-id="increase-button"]')
    button.trigger('click')
    expect(wrapper.emitted()['on-amount-changed']).toContainEqual([
      {
        amount: 3,
        size: 'small',
      },
    ])
  })
})
