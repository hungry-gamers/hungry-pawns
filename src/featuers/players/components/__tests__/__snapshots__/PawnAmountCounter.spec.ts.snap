// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PawnAmountCounter > should render properly 1`] = `"<div><span>small</span><button data-test-id="decrease-button"> - </button><span>2</span><button data-test-id="increase-button"> + </button></div>"`;
